plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c

```diff
@@ -0,0 +1,522 @@
+	if (!fu_logitech_bulkcontroller_device_send(self, msg1, BULK_INTERFACE_SYNC, error))
+		return FALSE;
+
+	/* wait for the response */
+	g_usleep(G_USEC_PER_SEC);
+	fu_byte_array_set_size(msg2, MAX_DATA_SIZE);
+	if (!fu_logitech_bulkcontroller_device_recv(self, msg2, BULK_INTERFACE_SYNC, 5000, error))
+		return FALSE;
+	msg3 = proto_manager_decode_message(msg2->data, msg2->len, &proto_id, error);
+	if (msg3 == NULL)
+		return FALSE;
+	if (proto_id != kProtoId_GetDeviceInfoResponse) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get kProtoId_GetDeviceInfoResponse");
+		return FALSE;
+	}
+
+	/* parse JSON reply */
+	if (!json_parser_load_from_data(json_parser, (const gchar *)msg3->data, msg3->len, error)) {
+		g_prefix_error(error, "error in parsing json data: ");
+		return FALSE;
+	}
+	json_root = json_parser_get_root(json_parser);
+	if (json_root == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON root");
+		return FALSE;
+	}
+	json_object = json_node_get_object(json_root);
+	json_payload = json_object_get_object_member(json_object, "payload");
+	if (json_payload == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON payload");
+		return FALSE;
+	}
+	json_devices = json_object_get_array_member(json_payload, "devices");
+	if (json_devices == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON devices");
+		return FALSE;
+	}
+	json_device = json_array_get_object_element(json_devices, 0);
+	if (json_device == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON device");
+		return FALSE;
+	}
+	if (json_object_has_member(json_device, "name"))
+		fu_device_set_name(device, json_object_get_string_member(json_device, "name"));
+	if (json_object_has_member(json_device, "sw"))
+		fu_device_set_version(device, json_object_get_string_member(json_device, "sw"));
+	if (json_object_has_member(json_device, "type"))
+		fu_device_add_instance_id(device,
+					  json_object_get_string_member(json_device, "type"));
+	if (json_object_has_member(json_device, "status"))
+		self->status = json_object_get_int_member(json_device, "status");
+	if (json_object_has_member(json_device, "updateStatus"))
+		self->update_status = json_object_get_int_member(json_device, "updateStatus");
+
+	/* success */
+	return TRUE;
+}
+
+static void
+fu_logitech_bulkcontroller_device_init(FuLogitechBulkcontrollerDevice *self)
+{
+	fu_device_add_protocol(FU_DEVICE(self), "com.logitech.vc.proto");
+	fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_TRIPLET);
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_UPDATABLE);
+	fu_device_register_private_flag(FU_DEVICE(self),
+					FU_LOGITECH_BULKCONTROLLER_DEVICE_FLAG_IS_MINI,
+					"is-mini");
```

Comment:
as I can tell this flag isn't actually used anywhere, can it go?  Or is it useful for debugging purposes?

Reply:
I don't believe we need it. 
fu_logitech_bulkcontroller_device_write_firmware() & fu_logitech_bulkcontroller_device_setup() are missing payload headers, and message exchange with device. I am working on it. Will keep you posted.

Reply:
I dropped it as a fixup commit, we can squash it when you have everything working.